misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordConverter.java

```diff
@@ -114,7 +115,7 @@ public static XMLToRecordResponse convertXMLToRecord(String xmlValue, boolean is
 
             Element rootElement = doc.getDocumentElement();
             generateRecords(rootElement, isClosed, recordToTypeDescNodes, recordToAnnotationNodes,
-                    recordToElementNodes, diagnosticMessages, value, withNameSpaces);
+                    recordToElementNodes, diagnosticMessages, fieldNameForLeafAttribute, withNameSpaces);
```

Comment:
instead of `fieldNameForLeafAttribute`, shall use `textFieldName` which is used in the apis options?

Reply:
+1